javascriptlib/src/mill/javascriptlib/PublishModule.scala

```diff
@@ -0,0 +1,539 @@
+package mill.javascriptlib
+
+import mill.*
+import os.*
+
+trait PublishModule extends TypeScriptModule {
+
+  /**
+   * Metadata about your project, required to publish.
+   *
+   * This is an equivalent of a `package.json`
+   */
+  def publishMeta: T[PublishModule.PublishMeta]
+
+  override def npmDevDeps: T[Seq[String]] =
+    Task { Seq("glob@^10.4.5", "ts-patch@3.3.0", "typescript-transform-paths@3.5.3") }
+
+  def bundledOut: T[String] = Task { "dist" }
+
+  private def declarationOut: T[String] = Task { "declarations" }
+
+  override def mainFileName: T[String] = Task { s"${millSourcePath.last}.js" }
```

Comment:
This doesn't seem related to publishing; should it be a default in the upstream `TypescriptModule`? Or should we leave it for the user to configure?

Reply:
- the mainFileName is from the `TscModule`
- it should be here, we are simply changing ts to js, as we publish js code.
- the user can change the mainFileName as always